tests/test_filter_count.py

```diff
@@ -180,39 +278,74 @@ def test_prune_on_small_data(filter_by):
     assert not any(noise_idx)
 
 
-def test_calibrate_joint():
+@pytest.mark.parametrize(
+    "multi_label",
+    [True, False],
+)
+def test_calibrate_joint(multi_label):
+    dataset = multilabel_data if multi_label else data
     cj = count.compute_confident_joint(
-        labels=data["labels"],
-        pred_probs=data["pred_probs"],
+        labels=dataset["labels"],
+        pred_probs=dataset["pred_probs"],
+        multi_label=multi_label,
         calibrate=False,
     )
     calibrated_cj = count.calibrate_confident_joint(
         confident_joint=cj,
-        labels=data["labels"],
+        labels=dataset["labels"],
+        multi_label=multi_label,
     )
-    label_counts = np.bincount(data["labels"])
 
     # Check calibration
-    assert all(calibrated_cj.sum(axis=1).round().astype(int) == label_counts)
-    assert len(data["labels"]) == int(round(np.sum(calibrated_cj)))
-
-    calibrated_cj2 = count.compute_confident_joint(
-        labels=data["labels"],
-        pred_probs=data["pred_probs"],
-        calibrate=True,
-    )
+    if multi_label:
+        y_one = int2onehot(dataset["labels"])
+        for class_num in range(0, len(calibrated_cj)):
+            label_counts = np.bincount(y_one[:, class_num])
+            assert all(calibrated_cj[class_num].sum(axis=1).round().astype(int) == label_counts)
+            assert len(dataset["labels"]) == int(round(np.sum(calibrated_cj[class_num])))
+        calibrated_cj2 = count.compute_confident_joint(
+            labels=dataset["labels"],
+            pred_probs=dataset["pred_probs"],
+            calibrate=True,
+            multi_label=multi_label,
+        )
+    else:
+        label_counts = np.bincount(data["labels"])
+        assert all(calibrated_cj.sum(axis=1).round().astype(int) == label_counts)
+        assert len(dataset["labels"]) == int(round(np.sum(calibrated_cj)))
+
+        calibrated_cj2 = count.compute_confident_joint(
+            labels=dataset["labels"],
+            pred_probs=dataset["pred_probs"],
+            calibrate=True,
+        )
 
     # Check equivalency
     assert np.all(calibrated_cj == calibrated_cj2)
 
 
 @pytest.mark.parametrize("use_confident_joint", [True, False])
-def test_estimate_joint(use_confident_joint):
-    joint = count.estimate_joint(
-        labels=data["labels"],
-        pred_probs=data["pred_probs"],
-        confident_joint=data["cj"] if use_confident_joint else None,
-    )
+@pytest.mark.parametrize("multi_label", [True, False])
+def test_estimate_joint(use_confident_joint, multi_label):
```

Comment:
same here, let's split this into two separate functions, one for multi_label one for regular classification. There is little purpose to have one function that encompasses two branches of logic when the two branches share little common code.